### PR TITLE
feat(flutterfire_ui): Add Turkish language localization support.

### PR DESCRIPTION
**Pull request by bgoktugozdemir**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by russellwheatley** [comment]
Hey @bgoktugozdemir , thanks for the PR! I don't suppose you are able to have your Turkish double checked by someone who speaks the language (i.e get this PR approved by a developer who speaks the language)? Just for spellchecking and correct language interpretation. I'm afraid I don't know anyone from Turkey but I imagine you do 😄

**Comment by bgoktugozdemir** [comment]
Hey @russellwheatley,

Can anyone who speaks Turkish check the PR or does the reviewer have to be a collaborator?

**Comment by russellwheatley** [comment]
I'm happy with a developer you know who can speak Turkish, it's just a proofreading exercise 😄 . Thanks.

File: packages/flutterfire_ui/lib/src/i10n/lang/tr.dart [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+  final String latitudeLabel;
+  @override
+  final String longitudeLabel;
+  @override
+  final String geopointLabel;
+  @override
+  final String referenceLabel;
+
+  const TrLocalizations({
+    this.emailInputLabel = 'Email',
```
Comment (salihgueler) [inline packages/flutterfire_ui/lib/src/i10n/lang/tr.dart:172]: E-mail

File: packages/flutterfire_ui/lib/src/i10n/lang/tr.dart [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    this.passwordInputLabel = 'Şifre',
+    this.signInActionText = 'Giriş yap',
+    this.registerActionText = 'Kayıt ol',
+    this.signInButtonText = 'Giriş yap',
+    this.registerButtonText = 'Kayıt ol',
+    this.linkEmailButtonText = 'İleri',
```
Comment (salihgueler) [inline packages/flutterfire_ui/lib/src/i10n/lang/tr.dart:178]: This says email button but the next shows translation for "Next". Is this expeceted?

File: packages/flutterfire_ui/lib/src/i10n/lang/tr.dart [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    this.signInWithAppleButtonText = 'Apple ile giriş yap',
+    this.signInWithTwitterButtonText = 'Twitter ile giriş yap',
+    this.signInWithFacebookButtonText = 'Facebook ile giriş yap',
+    this.phoneVerificationViewTitleText = 'Telefon numaranızı girin',
+    this.verifyPhoneNumberButtonText = 'İleri',
+    this.verifyCodeButtonText = 'Doğrula',
+    this.verifyingPhoneNumberViewTitle = 'SMS ile gelen kodu girin',
+    this.unknownError = 'Bilinmeyen bir hata meydana geldi',
+    this.smsAutoresolutionFailedError =
+        'SMS kodu otomatik olarak çözülemedi. Lütfen kodu manuel olarak girin',
```
Comment (salihgueler) [inline packages/flutterfire_ui/lib/src/i10n/lang/tr.dart:190]: @bgoktugozdemir Cozulemedi burada tam dogru degil gibi. Muadili uygulamalarda, "SMS kodu otomatik olarak eklenemedi" tarzinda bir kullanim var.

File: packages/flutterfire_ui/lib/src/i10n/lang/tr.dart [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    this.smsCodeSentText = 'SMS kodu gönderildi',
+    this.sendingSMSCodeText = 'SMS kodu gönderiliyor...',
+    this.verifyingSMSCodeText = 'SMS kodu doğrulanıyor...',
+    this.enterSMSCodeText = 'SMS kodunu girin',
+    this.emailIsRequiredErrorText = 'Email gerekli',
```
Comment (salihgueler) [inline packages/flutterfire_ui/lib/src/i10n/lang/tr.dart:195]: E-mail

File: packages/flutterfire_ui/lib/src/i10n/lang/tr.dart [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+    this.isNotAValidEmailErrorText = 'Geçerli bir email girin',
```
Comment (salihgueler) [inline packages/flutterfire_ui/lib/src/i10n/lang/tr.dart:196]: e-mail (bence e-posta hatta)

**Review by salihgueler — COMMENTED** [review]
I have couple of comments some of them are in English but most in Turkish. @bgoktugozdemir can you check them?

File: packages/flutterfire_ui/lib/src/i10n/lang/tr.dart [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+    this.chooseACountry = 'Bir ülke seçin',
+    this.enableMoreSignInMethods = 'Daha fazla oturum açma yöntemi etkinleştir',
+    this.signInMethods = 'Oturum açma yöntemleri',
+    this.provideEmail = 'Email ve şifrenizi girin',
+    this.goBackButtonLabel = 'Geri git',
+    this.passwordResetEmailSentText =
+        'Şifrenizi sıfırlamak için bir linki size email ile gönderdik. Lütfen emailinizi kontrol edin.',
+    this.forgotPasswordHintText =
+        'Emailinizi verin ve size şifrenizi sıfırlamanız için bir bağlantı gönderelim',
+    this.emailLinkSignInButtonLabel = 'Sihirli link ile giriş yap',
```
[the remark this answers is not in the document]
Comment (bgoktugozdemir) [reply packages/flutterfire_ui/lib/src/i10n/lang/tr.dart:239]: Türkiye'de Sihirli Bağlantı ve Magic Link olarak kullanılmış.

File: packages/flutterfire_ui/lib/src/i10n/lang/tr.dart (hunk 2 above)
Comment (bgoktugozdemir) [reply packages/flutterfire_ui/lib/src/i10n/lang/tr.dart:178]: The English version was:
```
this.linkEmailButtonText = 'Next',
```

File: packages/flutterfire_ui/lib/src/i10n/lang/tr.dart [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+    this.phoneInputLabel = 'Telefon numarası',
+    this.phoneNumberInvalidErrorText = 'Telefon numarası geçersiz',
+    this.phoneNumberIsRequiredErrorText = 'Telefon numarası gerekli',
+    this.profile = 'Profil',
+    this.name = 'İsim',
+    this.deleteAccount = 'Hesabı sil',
+    this.passwordIsRequiredErrorText = 'Şifre gerekli',
+    this.confirmPasswordIsRequiredErrorText = 'Şifrenizi onaylayın',
+    this.confirmPasswordDoesNotMatchErrorText = 'Şifreler uyuşmadı',
+    this.confirmPasswordInputLabel = 'Şifreyi onayla',
```
[the remark this answers is not in the document]
Comment (bgoktugozdemir) [reply packages/flutterfire_ui/lib/src/i10n/lang/tr.dart:217]: Yukarıdaki error mesajı. Dolayısıyla `Şifrenizi onaylayın` yazdım.

Aşağıdaki input label olduğu için `Şifreyi onayla` demiştim. `Şifreni onayla` olarak güncelledim.

File: packages/flutterfire_ui/lib/src/i10n/lang/tr.dart [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+    this.forgotPasswordButtonLabel = 'Şifrenizi mi unuttunuz?',
+    this.forgotPasswordViewTitle = 'Şifremi unuttum',
+    this.resetPasswordButtonLabel = 'Şifreyi sıfırla',
+    this.verifyItsYouText = 'Siz olduğunuzu doğrulayın',
+    this.differentMethodsSignInTitleText =
+        'Giriş yapmak için aşağıdaki yöntemlerden birini kullanın',
+    this.findProviderForEmailTitleText =
+        'Devam etmek için email adresinizi girin',
+    this.continueText = 'Devam et',
+    this.countryCode = 'Kod',
```
[the remark this answers is not in the document]
Comment (bgoktugozdemir) [reply packages/flutterfire_ui/lib/src/i10n/lang/tr.dart:227]: İngilizce halinde aşağıdaki gibi yazdığı için kod yazdım.

```
this.countryCode = 'Code',
```

File: packages/flutterfire_ui/lib/src/i10n/lang/tr.dart [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+    this.codeRequiredErrorText = 'Ülke kodu gerekli',
+    this.invalidCountryCode = 'Geçersiz kod',
```
[the remark this answers is not in the document]
Comment (bgoktugozdemir) [reply packages/flutterfire_ui/lib/src/i10n/lang/tr.dart:229]: İngilizce halinde aşağıdaki gibi yazdığı için Geçersiz kod yazdım.

```
this.invalidCountryCode = 'Invalid code',
```

**Review by salihgueler — APPROVED** [review]

**Comment by bgoktugozdemir** [comment]
@russellwheatley I guess everything is okay now. Can you check it again?

**Review by russellwheatley — APPROVED** [review]
LGTM

**Pull request merged**